src/FSLibrary/StellarStatefulSets.fs

```diff
@@ -371,3 +425,340 @@ type StellarFormation with
         // Final check after the loop completes
         if List.exists (fun (peer: Peer) -> peer.IsLoadGenComplete() = Failure) loadGenPeers then
             failwith "Loadgen failed!"
+
+    // Runs a P2P network infrastructure benchmark that mirrors the stellar-core network topology.
+    //
+    // Architecture Overview:
+    // - Creates a benchmark pod for each stellar-core node in the network with same connection topology of the actual stellar-core mission
+    // - Each benchmark pod runs in a StatefulSet (1 replica each) for stable DNS names
+    // - All StatefulSets share a single headless service for DNS resolution
+    // - Pod naming: {runId}-benchmark-{shortName}-0 (e.g., ssc-1234z-benchmark-lo-0-0)
+    //
+    // Pod Structure:
+    // Each benchmark pod contains two containers:
+    // 1. Server container: Runs multiple iperf3 servers
+    //    - One server per incoming connection from other nodes
+    //    - Each server listens on port 5201 + source_node_index (ensures unique ports)
+    // 2. Client container: Runs iperf3 clients
+    //    - Connects to all peer nodes as defined in the stellar-core topology
+    //    - iperf3 sends the maximum possible traffic to each peer node simultaneously
+    //    - Raw results saved to /results/*.json in the container
+    //
+    // Results Collection:
+    // - After tests complete, collects logs and raw iperf3 JSON results from all pods
+    // - parse_benchmark_results.py creates and writes a results summary file
+    member self.RunP2PNetworkBenchmark() : unit =
+        assert (self.NetworkCfg.missionContext.benchmarkInfrastructure.IsSome)
```

Comment:
Nit: this function would be more readable if it were broken up into several sub-functions